### PR TITLE
Codify labels and annotations that have been added directly to istio/istio

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/43437
Fixes https://github.com/istio/istio/issues/51575

### CODE REVIEW

File: annotation/annotations.yaml [hunk 1]
```diff
@@ -529,4 +529,54 @@ annotations:
...
     resources:
-      - Any
\ No newline at end of file
+      - Any
+
+  - name: ambient.istio.io/redirection
+    featureStatus: Beta
+    description: |-
+      Automatically configured by Istio to indicate a Pod was successfully enrolled in ambient mode.
+      This shows the actual state; to specify intent that a workload should be in ambient mode, see `istio.io/dataplane-mode`.
```
Comment (linsun) [inline annotation/annotations.yaml:538]: Should we add a note this should not be configured by a user?

**Review by linsun — COMMENTED** [review]
LGTM overall, just a minor comment

File: annotation/annotations.yaml (hunk 1 above)
Comment (howardjohn) [reply annotation/annotations.yaml:538]: thansk

**Comment by howardjohn** [comment]
@linsun done, thanks!

**Review by keithmattix — APPROVED** [review]

File: label/labels.yaml [hunk 2]
```diff
@@ -151,3 +159,90 @@ labels:
...
+    featureStatus: Beta
+    description: |
+      When set on a resource, indicates the resource has an associated waypoint in the provided namespace.
+      This must be set in addition to `istio.io/use-waypoint`, when a cross-namespace reference is desired.
+    deprecated: false
+    hidden: false
+    resources:
+      - Pod
+      - Service
+      - Namespace
```
Comment (ilrudie) [inline label/labels.yaml:224]: use-waypoint and use-waypoint-namespace also work on serviceEntry resources

**Review by linsun — APPROVED** [review]
LGTM, @ilrudie had a good comment, so i'll label this don't merge and you can merge it when you resolve his comment.

**Pull request merged**